### DISCUSSION TITLE
blog-heatmap/

### DISCUSSION THREAD

**Discussion by giscus** [eallion/eallion.com#133] [Giscus]
# blog-heatmap/

本文介绍了使用 CSS 和 JS 实现博客热力图的方法。通过获取最近一年的文章数据，并根据字数和发布日期生成相应的小方块，不同颜色代表不同级别的字数。同时，还添加了鼠标悬停时显示文章标题、字数等信息的功能。作者对比了几种类似库，并选择了适合自己需求的 heatmap.js 来实现热力图效果。最后给出了完整代码以及 HTML 和 CSS 样式部分。

https://www.eallion.com/blog-heatmap/

**Comment by flashlab** [comment]
[我也弄了一个heatmap](https://blog.zzbd.org/cals/)，不过是基于svg的，优点是dom更小，缺点是tooltip实现起来不方便 :persevere:

**Reply by eallion** [reply]
这个方法也研究过，确实各有各的优缺点。挺好看的。

**Comment by li199-code** [comment]
[https://blog.jasonleehere.com/archives/](我博客的归档页面)就是用您提供的方法实现的，留个言表示感谢！

**Reply by li199-code** [reply]
上面格式错了，[这个是正确链接](https://blog.jasonleehere.com/archives/)

**Reply by eallion** [reply]
谢谢关注！
